[Merge-Queue] Skip CloseBug if no bug defined

JonWBedard opened the pull request:
#### 6ddf72853c64be73426bce0a14fafa00690b3499
<pre>
[Merge-Queue] Skip CloseBug if no bug defined
<a href="https://bugs.webkit.org/show_bug.cgi?id=238612">https://bugs.webkit.org/show_bug.cgi?id=238612</a>
&lt;rdar://problem/91110116 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(CloseBug.start): Ignore unavailable bug_id case.
(CloseBug.getResultSummary): Use default summary for skip case.
(CloseBug.doStepIf): Do step if bug_id is available.
(CloseBug.hideStepIf): Hide step if skipped.

Canonical link: <a href="https://commits.webkit.org/249068@main">https://commits.webkit.org/249068@main</a>

From: Jonathan Bedard &lt;jbedard@apple.com &gt;


Canonical link: <a href="https://commits.webkit.org/249068@main">https://commits.webkit.org/249068@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292161">https://svn.webkit.org/repository/webkit/trunk@292161</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1720,23 +1720,29 @@ class CloseBug(buildstep.BuildStep, BugzillaMixin):
-            self.descriptionDone = 'No bug id found'
-            self.finished(FAILURE)
-            return None
+    def __init__(self, **kwargs):
```
seems like this __init__ can be removed, can initialize bug_id as class variable or can handle it appropriately in getResultsummary as well. But this is fine as well.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1720,23 +1720,29 @@ class CloseBug(buildstep.BuildStep, BugzillaMixin):
         return None
 
     def getResultSummary(self):
+        if self.results == SKIPPED:
```
Actually it's better not to have this custom result summary for SKIPPED case. First of all it wouldn't be displayed in the buildbot UI, since we set hideStepIf as well. Secondly, this custom string will show up in status-bubble pop-over message, while the default one (ending in `(skipped)`)  is hidden (See: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-app/ews/views/statusbubble.py#L70)

aj062 reviewed (approved):
r+ with comments.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1720,23 +1720,29 @@ class CloseBug(buildstep.BuildStep, BugzillaMixin):
+            return {'step': 'No bug to close'}
         if self.results == SUCCESS:
             return {'step': 'Closed bug {}'.format(self.bug_id)}
         return {'step': 'Failed to close bug {}'.format(self.bug_id)}
```
Actually, I missed noticing this default case handling (in this and previous few PRs). It doesn't look good. You should explicitly handle the cases which you want, and let buildbot default getResultSummary() handle the rest.

e.g.: https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L882 or https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L2834

JonWBedard replied:
Thanks for the explanation here, will be updating these later today.

JonWBedard committed:
[Merge-Queue] Skip CloseBug if no bug defined
https://bugs.webkit.org/show_bug.cgi?id=238612
<rdar://problem/91110116>

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(CloseBug.start): Ignore unavailable bug_id case.
(CloseBug.getResultSummary): Use default summary for skip case.
(CloseBug.doStepIf): Do step if bug_id is available.
(CloseBug.hideStepIf): Hide step if skipped.
